src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -351,24 +353,18 @@ private static boolean containsSummaryTag(DetailNode javadoc) {
      * @param javadoc javadoc root node.
      * @return first inline tag node or null if no node is found.
      */
-    private static DetailNode getFirstInlineTag(DetailNode javadoc) {
-        DetailNode node = null;
-        final DetailNode[] children = javadoc.getChildren();
-        for (DetailNode child: children) {
-            // If present as a children of javadoc
-            if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
-                node = child;
-            }
-            // If nested inside html tag
-            else if (child.getType() == JavadocTokenTypes.HTML_ELEMENT) {
-                node = getInlineTagNodeWithinHtmlElement(child);
-            }
+    private static Optional<DetailNode> getFirstInlineTag(DetailNode javadoc) {
+        final Predicate<DetailNode> childPredicate = child -> {
```

Comment:
Predicates are difficult to reuse. Please extract this predicate to a separated method.
This method, `getFirstInlineTag', on the contrary, can be inlined into `containsSummaryTag`.

Reply:
Done